### PR TITLE
systemd: add slice units

### PR DESCRIPTION
**Pull request by sashko**
Add a bunch of slice units for the future use.

Signed-off-by: Oleksandr Kravchuk <oleksandr.kravchuk@pelagicore.com>

### CODE REVIEW

File: recipes-core/systemd/systemd/dlt.slice [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[Unit]
+Description=GENIVI Diagnostic Log and Trace Slice
```
Comment (JEderonn) [inline recipes-core/systemd/systemd/dlt.slice]: Remove word GENIVI since it would be possible to put other than GENIVI services into this slice.

**Comment by JEderonn** [comment]
Confirm that it's correct to have this PR in meta-bistro and not in meta-pelux.

The set of slice units and the assignment of services to slice units are product specific, so we need to provide a solution where a customer can do its own configuration. For example, move a service which has been assigned to slice media.slice to another custom specific slice which hasn't been defined/added in meta-bistro.

**Comment by sashko** [comment]
> ...For example, move a service which has been assigned to slice media.slice to another custom specific slice which hasn't been defined/added in meta-bistro.

One can modify, replace or completely remove service and slice unit files with bbappends.

**Comment by JEderonn** [comment]
> One can modify, replace or completely remove service and slice unit files with bbappends.

Sure, this can be done.

**Comment by JEderonn** [comment]
I'm sorry, but you didn't have to add system.slice since this woujld be the default slice unit for those services which are not explicitly assigned to anything. Just checked it on target.

**Comment by sashko** [comment]
@JEderonn why was this closed? :confused:

**Comment by JEderonn** [comment]
> @JEderonn why was this closed? 😕

Hmm, my mistake, sorry. It was not my intention to close the PR.

**Comment by martin-ejdestig** [comment]
> > ...For example, move a service which has been assigned to slice media.slice to another custom specific slice which hasn't been defined/added in meta-bistro.
> 
> One can modify, replace or completely remove service and slice unit files with bbappends.

I do not know for sure but when first seeing this it initially seemed to me like this belonged in meta-pelux. bbappend:ing a bbappend seems narly. One also has to know explicitly which slices to remove? E.g. if updating to a meta-bistro that has added yet another slice it will not be detected without explicitly checking and sucked in to the product if it just happens to use a component from meta-bistro?

**Comment by sashko** [comment]
> I do not know for sure but when first seeing this it initially seemed to me like this belonged in meta-pelux. bbappend:ing a bbappend seems narly. 

I would disagree: `meta-pelux`'s function is to provide an extension to `meta-vistro` with Qt Automotive Suit and Neptune 3 UI. All the system-level is a job of `distro` bistro.

> One also has to know explicitly which slices to remove? E.g. if updating to a meta-bistro that has added yet another slice it will not be detected without explicitly checking and sucked in to the product if it just happens to use a component from meta-bistro?

We cannot predict what a user might want/need. Our job is to provide a framework and examples. If nothing suits them, they can bbmask our bbappend - this is a common practice in Yocto.

**Comment by JEderonn** [comment]
> We cannot predict what a user might want/need. Our job is to provide a framework and examples. If nothing suits them, they can bbmask our bbappend - this is a common practice in Yocto.

I think this is the way to go. meta-bistro will provide the framework and include the default setup. The customer can then choose to reconfigure if needed by bbappend in meta-<customer>.

**Comment by sashko** [comment]
> meta-bistro

just double-checking, `meta-bistro`?

**Comment by BassemMohsen** [comment]
> > meta-bistro
> 
> just double-checking, `meta-bistro`?

I would argue, It should be in meta-pelux. I don't see why do we need an extra layer like meta-bistro at the first place.

The slice units would have to be referenced by the services in meta-pelux. therefore it should be shipped in the same layer to be modular functionality when updating or changing services and slices.
I guess this what @martin-ejdestig  has mentioned above.

**Comment by martin-ejdestig** [comment]
Regardless of whether we put this in meta-bistro or meta-pelux, the names of the slices look wrong. Or has this:

"Slices are organized hierarchically in a tree. The name of the slice encodes the location in the tree. The name consists of a dash-separated series of names, which describes the path to the slice from the root slice. The root slice is named -.slice. Example: foo-bar.slice is a slice that is located within foo.slice, which in turn is located in the root slice -.slice."

from https://www.freedesktop.org/software/systemd/man/systemd.slice.html been taken into account?

**Pull request closed without merging**